internal/ir/xds.go

```diff
@@ -108,6 +109,9 @@ func (h HTTPListener) Validate() error {
 // TLSListenerConfig holds the configuration for downstream TLS context.
 // +k8s:deepcopy-gen=true
 type TLSListenerConfig struct {
+	// TLSMode could either be Terminate or Passthrough. If the TLS mode
+	// is Passthrough, certificate and private key information is not required.
+	TLSMode v1beta1.TLSModeType
```

Comment:
I think TLS passthrough warrants its own Listener e.g.
```

type Xds struct {
	// HTTP listeners exposed by the gateway.
	HTTP []*HTTPListener
	// TLS passthrough listener exposed by the gateway
	TLS []*TLSListener
}
```

Reply:
I started off with that, wasn't too sure about it so just added the tls mode in HTTPListener, assuming HTTPListener could just be called and used as a generic Listener (similar to the gateway object's listener).

maybe adding a different tls route makes more sense?
```
type HTTPListener struct {  // rename to "Listener"
    Name, Address, Port, Hostnames, TLS
    
    Routes  []*HTTPRoute
    TLSRoutes []*TLSRoute
}
```

Reply:
I prefer if you created a new one because the fields within `HTTPRoute` dont apply to TLS passthrough

Reply:
done. thanks